### ISSUE TITLE
RGCN demo gives "Gradients do not exist for variables ['relational_graph_convolution_1/bases:0' ..." for every epoch

### ISSUE THREAD

**Issue by habiba-h** [stellargraph/stellargraph#649]
### Description

The RGCN demo has tensorflow warnings about gradients not existing when training the model, e.g.

```
Epoch 1/20
WARNING:tensorflow:From /Users/wil9dw/.pyenv/versions/sg4/lib/python3.6/site-packages/tensorflow_core/python/ops/array_grad.py:502: _EagerTensorBase.cpu (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.identity instead.
WARNING:tensorflow:Gradients do not exist for variables ['relational_graph_convolution_1/bases:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_3/bases:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0'] when minimizing the loss.
1/1 [==============================] - 16s 16s/step - loss: 1.6101 - acc: 0.2746 - val_loss: 1.5919 - val_acc: 0.3333
Epoch 2/20
WARNING:tensorflow:Gradients do not exist for variables ['relational_graph_convolution_1/bases:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_3/bases:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0'] when minimizing the loss.
...
```

**Original description:**

`RelationalFullBatchNodeGenerator` is not working as it should, very likely this is where the problem occurs:
```
for edge_type in edge_types:

            col_index = [
                self.node_index[n1]
                for n1, n2, etype in G.edges(triple=True)
                if etype == edge_type
            ]
            row_index = [
                self.node_index[n2]
                for n1, n2, etype in G.edges(triple=True)
                if etype == edge_type
            ]
```

This is due to the `G.edges(triple=True)` method which returns as `edge_type` the index in the dictionary of edges and not the actual `edge_type`. 

The `RelationalFullBatchNodeGenerator` does not break down as it is still getting something for the `edge_type` however, its not what it should be. 
Training the `RGCN` results in  absence of gradients for a number of variables in the model. 
To reproduce this issue run the `RGCN`: [demo](https://github.com/stellargraph/stellargraph/blob/develop/demos/node-classification/rgcn/rgcn-aifb-node-classification-example.ipynb)
### User Story
As an: ML practitioner
I want: use `RGCN` model implemented in `StellarGraph` library
so that: I can do node classification on my data.

### Done Checklist (Development)
- [ ] Assumptions of the user story met
- [ ] Produced code for required functionality
- [ ] Branch and Pull Request build on CI
- [ ] Branch and Pull Request pass unit tests on CI
- [ ] Branch and Pull Request pass integration tests on CI
- [ ] Version number reflects new status
- [ ] Peer Code Review Performed
- [ ] Code well commented
- [ ] Documentation on Google Docs
- [ ] Documentation in repo
- [ ] CHANGELOG.md updated
- [ ] Team demo
- [ ] Mini-meetup talk
- [ ] Stakeholder sign-off

### Done Checklist (Bug)
- [ ] Bug fixed
- [ ] Branch and Pull Request build on CI
- [ ] Branch and Pull Request pass unit tests on CI
- [ ] Branch and Pull Request pass integration tests on CI
- [ ] Version number reflects new status
- [ ] Peer Code Review Performed
- [ ] Code well commented
- [ ] CHANGELOG.md updated

**Comment by huonw** [comment]
I think this is a duplicate of #642 (and thus will be fixed by #643) since it seems like the core problem is the behaviour of `edges(triple=True)`.

**Comment by huonw** [comment]
Hm, or maybe it isn't. I've updated the RGCN notebook in #643 and it still hits these errors.
